src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/DesignerOptionService.cs

```diff
@@ -226,29 +200,29 @@ public PropertyDescriptorCollection Properties
                 {
                     if (_properties == null)
                     {
-                        ArrayList propList;
+                        List<PropertyDescriptor> propList;
 
                         if (_value != null)
                         {
                             PropertyDescriptorCollection props = TypeDescriptor.GetProperties(_value);
-                            propList = new ArrayList(props.Count);
+                            propList = new List<PropertyDescriptor>(props.Count);
                             foreach (PropertyDescriptor prop in props)
                             {
                                 propList.Add(new WrappedPropertyDescriptor(prop, _value));
                             }
                         }
                         else
                         {
-                            propList = new ArrayList(1);
+                            propList = new List<PropertyDescriptor>(1);
                         }
 
                         EnsurePopulated();
                         foreach (DesignerOptionCollection child in _children)
                         {
-                            propList.AddRange(child.Properties);
+                            propList.AddRange(Enumerable.Cast<PropertyDescriptor>(child.Properties));
```

Comment:
This adds expense, at least an allocation, plus additional virtual method calls.  Probably best to revert these changes and stick with using ArrayList, since the input we're dealing with is itself non-generic.